### ISSUE TITLE
Search config (facet) not persisted

### ISSUE THREAD

**Issue by coret** [Daniel-KM/Omeka-S-module-AdvancedSearch#14]
When I try to add a facet to my search page (/admin/search-manager/config/4/configure#facet) by adding an facet, this information isn't stored. When I edit the configuration again the added facet is gone (and not visible on my search page).

My (advanced) search page: https://www.goudatijdmachine.nl/omeka/s/data/zoeken?q=raam

I am using [Advanced Search](https://gitlab.com/Daniel-KM/Omeka-S-module-AdvancedSearch) version 3.4.31 and [Advanced Search adapter for Solr](https://gitlab.com/Daniel-KM/Omeka-S-module-SearchSolr) version 3.5.47 on [Omeka S](https://omeka.org/s) version 4.1.1

**Comment by Daniel-KM** [comment]
Yes, there is a strange issue depending on the version of php : 7.4 works fine. Can you try version 3.4.32 ?

**Comment by coret** [comment]
I am nog running 3.4.32, but still can't persist facets of search page config of my solr instance. I see the following errors in the log table:
```
Creation of dynamic property AdvancedSearch\Controller\Admin\SearchConfigController::$formElementManager is deprecated {extra}

{"errno":8192,"file":"/home/http/goudatijdmachine.nl/omeka-s-modules/AdvancedSearch/src/Controller/Admin/SearchConfigController.php","line":70,"extra":"{\"errno\":8192,\"file\":\"/home/http/goudatijdmachine.nl/omeka-s-modules/AdvancedSearch/src/Controller/Admin/SearchConfigController.php\",\"line\":70}"}

And a lot of
Undefined array key "value" {extra}
{"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327,"extra":"{\"errno\":2,\"file\":\"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php\",\"line\":327}"}
```

**Comment by Daniel-KM** [comment]
The first deprecation warning is fixed in new commits after 3.4.32 (so will be released next week).
The second issue may be fixed in last commit too. Can you try it?

**Comment by coret** [comment]
I've updated the module with the current code from this repo:  less errors, but still can't save the search config.

BTW when I just open https://www.goudatijdmachine.nl/omeka/admin/search-manager/config/4/configure I see te following in the log:
```
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
2024-10-14T13:38:36+00:00 WARN (4): Undefined array key "value" {"errno":2,"file":"/home/http/goudatijdmachine.nl/omeka-s-4.1.1/vendor/laminas/laminas-form/src/Element/Select.php","line":327} {"userId":2}
```
I notice that I can change for example the **Label "No facets"** value on the Facets tab and this is persisted. But adding a facet via the **+** button (with Field dropdown set) is not persisted.
